### PR TITLE
Enable username&password passed in URL

### PR DESCRIPTION
**Pull request by mallexxx**
Please review the release process for BrowserServicesKit [here](https://app.asana.com/0/1200194497630846/1200837094583426).

**Required**:

Task/Issue URL: https://app.asana.com/0/414235014887631/1204080079383830/f
Tech Design URL: https://app.asana.com/0/481882893211075/1204065038916791/f
Security Triage URL: https://app.asana.com/0/1199892415909552/1204067721102886/f
iOS PR:  https://github.com/duckduckgo/iOS/pull/1521
macOS PR: https://github.com/duckduckgo/macos-browser/pull/1006
What kind of version bump will this require?: Minor

**Description**:
- Removed URL.isValid check returning false for URLs containing credentials
- Fixed parsing punycoded URLs with credentials and mailto: URLs

**Steps to test this PR**:
1. Validate platform PRs

**OS Testing**:

* [ ] iOS
* [ ] macOS

---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)

### CODE REVIEW

File: Sources/Common/Extensions/URLExtension.swift [hunk 1]
```diff
@@ -91,28 +91,54 @@ extension URL {
         public static let blob = NavigationalScheme(rawValue: "blob")
         public static let about = NavigationalScheme(rawValue: "about")
 
+        public static let mailto = NavigationalScheme(rawValue: "mailto")
```
Comment (brindy) [inline Sources/Common/Extensions/URLExtension.swift:94]: Are these changes related?

**Review by brindy — CHANGES_REQUESTED** [review]
Unsure what the mailto thing is about, but if that's deliberate this code looks OK and I've seen it working on iOS (under some conditions).

However, the tests appear to be failing.  If these are flakey tests, can you comment them out / remove them and create tasks for the owners, please?

File: Sources/Common/Extensions/URLExtension.swift (hunk 1 above)
Comment (mallexxx) [reply Sources/Common/Extensions/URLExtension.swift:94]: mailto uses the same user@domain.com URL structure which was returning false for url.isValid for punycoded URLs

**Comment by mallexxx** [comment]
the flaky unit tests are fixed in https://github.com/duckduckgo/BrowserServicesKit/pull/240

**Review by brindy — APPROVED** [review]
Tested macOS and iOS and all seems to work as expected. LGTM
